addons/sale_stock/models/sale_order.py

```diff
@@ -403,3 +403,16 @@ def _update_line_quantity(self, values):
             raise UserError(_('You cannot decrease the ordered quantity below the delivered quantity.\n'
                               'Create a return first.'))
         super(SaleOrderLine, self)._update_line_quantity(values)
+
+
+class ProductionLot(models.Model):
+    _inherit = 'stock.production.lot'
+
+    sale_order_ids = fields.Many2many('sale.order', string="Sale Orders", compute='_sale_orders', readonly=True, store=False)
+
+    @api.depends('name')
+    def _sale_orders(self):
+        for stock_move_line in self.env['stock.move.line'].search([('lot_id','=',self.id),('state','=','done')]):
+            for stock_move in stock_move_line.move_id:
+                if (stock_move_line.move_id.picking_id.location_dest_id.usage == "customer" and stock_move.state=="done" ):
+                    self.sale_order_ids = [ stock_move.sale_line_id.order_id.id]
```

Comment:
There's no new line at end of file. 

This override should be moved into stock.py, where all the stock-related objects are defined.

Reply:
A few more comments:
- The trigger is not correct. On the other hand, I'm not sure what we could set instead. 
- The method doesn't work in batch. It's working because the field is only displayed on a form view, but if you display it on a list view or if you call the method yourself on a recordset, it will crash.
- By doing ``self.sale_order_ids = [ stock_move.sale_line_id.order_id.id]`` you erase the result from the previous loop, which is not correct too.
- By doing ``for stock_move in stock_move_line.move_id:`` you will loop several times on the same stock_move.